types/big.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	"fmt"
+	"math/big"
+)
+
+// BigInt is a big.Int wrapper which marshals JSON to a string representation of the big number
+type BigInt big.Int
+
+func (i BigInt) MarshalText() ([]byte, error) {
+	return []byte((*big.Int)(&i).String()), nil
+}
+
+func (i *BigInt) UnmarshalText(data []byte) error {
+	i2, ok := new(big.Int).SetString(string(data), 0)
+	if !ok {
+		return fmt.Errorf("wrong format for bigInt")
+	}
+	*i = (BigInt)(*i2)
+	return nil
+}
+
+func (i *BigInt) GobEncode() ([]byte, error) {
+	return i.ToInt().GobEncode()
+}
+
+func (i *BigInt) GobDecode(buf []byte) error {
+	return i.ToInt().GobDecode(buf)
+}
+
+// String returns the string representation of the big number
+func (i *BigInt) String() string {
+	return fmt.Sprintf("%s", (*big.Int)(i))
+}
+
+// ToInt converts b to a big.Int.
+func (i *BigInt) ToInt() *big.Int {
+	return (*big.Int)(i)
+}
+
+// Add sum x+y
+func (i *BigInt) Add(x *BigInt, y *BigInt) *BigInt {
+	return (*BigInt)(i.ToInt().Add(x.ToInt(), y.ToInt()))
+}
+
+// SetUint64 sets the value of x to the big number
```

Comment:
I think this is going to be used a lot in tests create BigInt from literal integers.  And this way it's the same length as using a type cast:
```go
new(BigInt).SetUint64(100) // using BigInt method
(*BigInt)(big.NewInt(100)) // type cast
```
I suggest making this a constructor making its common usage shorter:
`func NewBigInt(x uint64) *BigInt` or 
Then we have:
```go
NewBigInt(100)
```

This is just a suggestion!  Feel free to ignore it.

Reply:
thanks for the suggestion, but I prefer to have the SetUint64() method as part of the type.